src/main/java/tfifteenfour/clipboard/model/course/Group.java

```diff
@@ -0,0 +1,81 @@
+package tfifteenfour.clipboard.model.course;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+
+import javafx.collections.ObservableList;
+import tfifteenfour.clipboard.model.student.Assignment;
+import tfifteenfour.clipboard.model.student.Session;
+import tfifteenfour.clipboard.model.student.Student;
+import tfifteenfour.clipboard.model.student.UniqueStudentList;
+
+public class Group {
+	public static final String MESSAGE_CONSTRAINTS = "Module codes should be alphanumeric";
+	public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+
+	public String groupName;
+	private final UniqueStudentList students;
+
+	{
+		students = new UniqueStudentList();
+	}
```

Comment:
Similar with `groups`, may I know why `students = new UniqueStudentList();` needs to be separated?